tools/libSwiftScan/CMakeLists.txt

```diff
@@ -30,14 +30,36 @@ set_target_properties(libSwiftScan
     PROPERTIES
     OUTPUT_NAME ${SWIFT_SCAN_LIB_NAME})
 
-if(SWIFT_SWIFT_PARSER)
-  # Ensure that we can find the host shared libraries.
-  set_property(
-    TARGET libSwiftScan
-    APPEND PROPERTY INSTALL_RPATH "@loader_path/swift/host")
+if(SWIFT_HOST_VARIANT_SDK MATCHES "LINUX|ANDROID|OPENBSD|FREEBSD" AND BOOTSTRAPPING_MODE STREQUAL "HOSTTOOLS")
+  # Runtime INSTALL_RPATH are set by 'add_swift_host_library', but that expects
+  # libSwiftScan be installed in 'lib'. But since it's actually installed in 'lib/swift/host',
+  # we need to have correct runtime path to 'lib/swift/{platform}'.
+  # FIXME: BUILD_RPATH and INSTALL_PATH should be different
+  # FIXME: add_swift_host_library should accept 'DESTINATION' and handle installation
+  # FIXME: Build this library into 'lib/swift/host/' instead of 'lib/'
   set_property(
     TARGET libSwiftScan
-    APPEND PROPERTY INSTALL_RPATH "@loader_path/../host")
+    APPEND PROPERTY INSTALL_RPATH "$ORIGIN/../${SWIFT_SDK_${SWIFT_HOST_VARIANT_SDK}_LIB_SUBDIR}"
+  )
+endif()
+
+if(SWIFT_SWIFT_PARSER)
+  if(SWIFT_HOST_VARIANT_SDK IN_LIST SWIFT_DARWIN_PLATFORMS)
+    # Ensure that we can find the host shared libraries.
+    set_property(
+      TARGET libSwiftScan
+      APPEND PROPERTY INSTALL_RPATH "@loader_path/swift/host")
+    set_property(
+      TARGET libSwiftScan
+      APPEND PROPERTY INSTALL_RPATH "@loader_path/../host")
+  elseif(SWIFT_HOST_VARIANT_SDK MATCHES "LINUX|ANDROID|OPENBSD|FREEBSD")
+    set_property(
+      TARGET libSwiftScan
+      APPEND PROPERTY INSTALL_RPATH "$ORIGIN/swift/host")
```

Comment:
@rintaro, is this needed for some intermediate build step or something? Because the final installed library also has this path, which is wrong:
```
>

Reply:
Yeah, see the many FIXME above. This all needs to be split into BUILD/INSTALL_RPATH + built into host rather than into lib, but that's too much for 5.9. We'll fix this up after all the 5.9 PRs are in as well. We could remove it, but it doesn't really hurt either.